### PR TITLE
Update paddlenlp snippet to use `transformersInfo`

### PR DESCRIPTION
**Pull request by sijunhe**
`PaddleNLP` has upgraded to re-use the same config system as `transformers`. Updating the paddlenlp snippet to produce a more accurate snippet by using the `transformersInfo`

### CODE REVIEW

File: js/src/lib/interfaces/Libraries.ts [hunk 1]
```diff
@@ -158,12 +158,17 @@ model = from_pretrained_keras("${model.id}")
 `;
 
 const paddlenlp = (model: ModelData) => {
-	return [
-	  `from paddlenlp.transformers import AutoModel, AutoTokenizer`,
-	  "",
-	  `tokenizer = AutoTokenizer.from_pretrained("${model.id}"${model.private ? ", use_auth_token=True" : ""}, from_hf_hub=True)`,
-	  `model = AutoModel.from_pretrained("${model.id}"${model.private ? ", use_auth_token=True" : ""}, from_hf_hub=True)`,
-	].join("\n");
+	const architecture = model.config.architectures[0];
```
Comment (xianbaoqian) [inline js/src/lib/interfaces/Libraries.ts:161]: Please also check if model.config exists.

Comment (xianbaoqian) [reply js/src/lib/interfaces/Libraries.ts:161]: image

**Review by xianbaoqian — COMMENTED** [review]
Looks good. :-)

**Comment by xianbaoqian** [comment]
Looks good. Thanks!

**Review by xianbaoqian — APPROVED** [review]

File: js/src/lib/interfaces/Libraries.ts [hunk 2]
```diff
@@ -158,12 +158,17 @@ model = from_pretrained_keras("${model.id}")
 `;
 
 const paddlenlp = (model: ModelData) => {
-	return [
-	  `from paddlenlp.transformers import AutoModel, AutoTokenizer`,
-	  "",
-	  `tokenizer = AutoTokenizer.from_pretrained("${model.id}"${model.private ? ", use_auth_token=True" : ""}, from_hf_hub=True)`,
-	  `model = AutoModel.from_pretrained("${model.id}"${model.private ? ", use_auth_token=True" : ""}, from_hf_hub=True)`,
-	].join("\n");
+	if (model.config && model.config.architectures && model.config.architectures[0]) {
```
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:161]: I think it's also possible to do

```suggestion
	if (model.config?.architectures?.[0]) {
```

File: js/src/lib/interfaces/Libraries.ts [hunk 3]
```diff
@@ -158,12 +158,17 @@ model = from_pretrained_keras("${model.id}")
...
+		const architecture = model.config.architectures[0];
+		return [
+			`from paddlenlp.transformers import AutoTokenizer, ${architecture}`,
+			"",
+			`tokenizer = AutoTokenizer.from_pretrained("${model.id}"${model.private ? ", use_auth_token=True" : ""}, from_hf_hub=True)`,
+			`model = ${architecture}.from_pretrained("${model.id}"${model.private ? ", use_auth_token=True" : ""}, from_hf_hub=True)`,
+		].join("\n");
+	} else {
+		return `# ⚠️ Type of model unknown`;
```
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:170]: Would `AutoModel` be ok here?

**Review by osanseviero — APPROVED** [review]
Thanks!

File: js/src/lib/interfaces/Libraries.ts (hunk 3 above)
Comment (osanseviero) [reply js/src/lib/interfaces/Libraries.ts:170]: Yes, exactly :)

**Comment by sijunhe** [comment]
Addressed the fall back comment. I think we are ready to merge.

**Review by osanseviero — APPROVED** [review]
Thanks!

**Pull request merged**